src/js/adapters/fileReader.test.js

```diff
@@ -115,14 +71,98 @@ describe('countLinesInBuffer', () => {
   });
 });
 
-describe('Fetching a buffer from a file', () => {
-  it('should not take too long', () => {
-    return readDataFromByte('src/resources/testFile', 100000, 1000).then(
-      data => {
-        console.log(data.lines);
-        console.log(data.linesStartAt);
-        console.log(data.linesEndAt);
-      }
+describe('readDataFromByte', () => {
+  //   it('should return an object with the expected content', async () => {
+  //     const filePath = 'src/resources/logFileForTests';
+  //     const startReadFromByte = 0;
+  //     const numberOfBytes = 500;
+  //     const expectedResult = {
+  //       startByteOfLines: [0, 115, 232, 344],
+  //       lines: [
+  //         '2020-02-03 14:50:17.916  INFO  16639 --- [        Thread1] o.s.d.r.w.RepositoryRestHandlerAdapter   : Changed data',
+  //         '2020-02-03 14:50:18.252  INFO  379   --- [        Thread2] o.s.b.w.embedded.tomcat.TomcatWebServer  : info message 1',
+  //         '2020-02-03 14:50:18.588  INFO  3675  --- [        Thread1] o.s.d.r.w.RepositoryRestHandlerMapping   : Forbidden',
+  //         '2020-02-03 14:50:18.926  INFO  12772 --- [        Thread1] o.s.d.r.w.RepositoryRestHandlerMapping   : Changed data'
+  //       ],
+  //       linesStartAt: 0,
+  //       linesEndAt: 463
+  //     };
+  //     const testResult = await readDataFromByte(
+  //       filePath,
+  //       startReadFromByte,
+  //       numberOfBytes
+  //     );
+
+  //     expect(testResult).toEqual(expectedResult);
+  //   });
+
+  // it('should not return first row when byte to read from starts after 0 and before new line', async () => {
+  //   const filePath = 'src/resources/logFileForTests';
+  //   const startReadFromByte = 20;
+  //   const numberOfBytes = 500;
+  //   const expectedResult = {
+  //     startByteOfLines: [115, 232, 344],
+  //     lines: [
+  //       '2020-02-03 14:50:18.252  INFO  379   --- [        Thread2] o.s.b.w.embedded.tomcat.TomcatWebServer  : info message 1',
+  //       '2020-02-03 14:50:18.588  INFO  3675  --- [        Thread1] o.s.d.r.w.RepositoryRestHandlerMapping   : Forbidden',
+  //       '2020-02-03 14:50:18.926  INFO  12772 --- [        Thread1] o.s.d.r.w.RepositoryRestHandlerMapping   : Changed data'
+  //     ],
+  //     linesStartAt: 115,
+  //     linesEndAt: 463
+  //   };
+  //   const testResult = await readDataFromByte(
+  //     filePath,
+  //     startReadFromByte,
+  //     numberOfBytes
+  //   );
+  //   expect(testResult.startByteOfLines[0]).not.toEqual(startReadFromByte);
+  //   expect(testResult).toEqual(expectedResult);
+  // });
```

Comment:
These tests could be uncommented. As well as the last one. Tried running them and they went through for me.

Reply:
Aha! I uncommented them and tried running, but they failed at a couple of places. I did some minor changes in them in the latest commit and they work now for me. Could you see if they still work for you?